generic-ec-zkp/src/polynomial.rs

```diff
@@ -400,7 +400,10 @@ mod tests {
         // 2. Commit to the secret
         // F(x) = G * f(x)
         let F = &f * &Point::generator();
-        assert_eq!(&secret * Point::generator(), F.value(&Scalar::zero()));
+        assert_eq!(
+            &secret * Point::generator(),
+            F.value::<_, Point<_>>(&Scalar::zero())
+        );
```

Comment:
Why did it change, can we fix it back? I hate explicit types

Reply:
I think it's because I implemented PartialEq for NonZero, and now when compiler sees `Point<E>` being compared with some generic `O`, it can't anymore deduce that it's `Point<E>` because it could be `NonZero<Point<E>>` as well